### PR TITLE
New Product Docs - First pass

### PR DESCRIPTION
**Pull request by paulb777**
#no-changelog

### CODE REVIEW

File: docs/AddNewPod.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## Introduction
 
+**This document is deprecated in favor of [docs/AddNewProduct.md](AddNewProduct.md)**. It
+provides additional context about legacy Objective C implementations.
+
```
Comment (ncooke3) [inline docs/AddNewPod.md:7]: optional nit: reduce to only one newline
```suggestion
```

File: docs/AddNewProduct.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+Firebase uses [swiftformat](https://github.com/nicklockwood/SwiftFormat) for enforcing code
+formatting consistency.
+
+## Package Managers
+
+Firebase supports four different distributions - Swift Package Manager, CocoaPods, Carthage, and
+binary zip distributions.
+
+It can be developed via Swift Package Manager or CocoaPods.
```
Comment (ncooke3) [inline docs/AddNewProduct.md:35]:
```suggestion
Firebase SDKs can be developed via Swift Package Manager or CocoaPods.
```

File: docs/AddNewProduct.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+The new project should set up a `podspec` for CocoaPods and add a `product` specification to
+[Package.swift](Package.swift).
```
Comment (ncooke3) [inline docs/AddNewProduct.md:38]:
```suggestion
the [Package.swift](Package.swift).
```

File: docs/AddNewProduct.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+All Firebase code should be unit tested. Ideally, the unit tests should be driven from both
+CocoaPods and Swift Package Manager. If only one is implemented, it should be SPM.
+
+## Dependencies
+
+Dependencies are a great way to add libraries of well-tested functionality to Firebase. On the flip
+side, they can add bloat and risk to the Firebase user experience. Ideally, only existing Firebase
+dependencies should be used. See [Package.swift](Package.swift). If a new dependency is needed,
+consider making it a weak dependency and only hard depending on a protocol. Any new hard
```
Comment (ncooke3) [inline docs/AddNewProduct.md:50]: I don't quite follow the _and only hard depending on a protocol_ part. Could you maybe rework the back half of that sentence?

**Review by ncooke3 — APPROVED** [review]
nice!

File: docs/AddNewProduct.md (hunk 4 above)
Comment (paulb777) [reply docs/AddNewProduct.md]: @ncooke3 PTAL

**Review by ncooke3 — APPROVED** [review]

**Pull request merged**